include/net/coap.h

```diff
@@ -231,6 +231,9 @@ struct coap_packet {
 	uint8_t hdr_len; /* CoAP header length */
 	uint16_t opt_len; /* Total options length (delta + len + value) */
 	uint16_t delta; /* Used for delta calculation in CoAP packet */
+#if defined(CONFIG_COAP_KEEP_SOCKET)
+	int sock; /* Incoming socket, used to send ACK reply */
```

Comment:
You introduce a new field in this struct but I see no one using it, are you missing some commits?